salesagility/SuiteCRM
# Fix #10491 - Display error message when importing user that it reports to itself

## Description
This PR checks that a user is not validating themselves or any other user(s) who end up reporting to them, so that a loop is not created in the chain of reporters.

## How To Test This
1. Create a normal user who reports to himself.
2. Export the user and in the downloaded document change the name of the user and the ID of the user to whom he reports ("Reports to" column), where we will indicate the ID of the user himself
3. Run the Import and Update process and check that error message is displayed and the user is not updated.

(Optional)
5. Create another user who reports to the first
6. Export both users and modify the first user so that it reports to this second user.
7. Run the Import and Update process and check that error message is displayed and the user is not updated.

+Optional
8. Extend the chain of reporters and perform the checks.

## Types of changes


### Final checklist
